crates/amaru/src/stages/consensus/forward_chain/tcp_forward_chain_server.rs

```diff
@@ -0,0 +1,211 @@
+// Copyright 2024 PRAGMA
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::stages::PallasPoint;
+use crate::stages::consensus::forward_chain::client_protocol::{
+    ClientMsg, ClientOp, ClientProtocolMsg, client_protocols,
+};
+use acto::{AcTokio, ActoCell, ActoMsgSuper, ActoRef, ActoRuntime, MailboxSize};
+use amaru_consensus::consensus::effects::network_effects::{ForwardEvent, ForwardEventListener};
+use amaru_consensus::consensus::tip::{AsHeaderTip, HeaderTip};
+use amaru_kernel::Header;
+use amaru_ouroboros_traits::ChainStore;
+use async_trait::async_trait;
+use pallas_network::{facades::PeerServer, miniprotocols::chainsync::Tip};
+use std::collections::BTreeMap;
+use std::sync::Arc;
+use std::sync::Mutex;
+use tokio::net::TcpListener;
+
+pub const EVENT_TARGET: &str = "amaru::consensus::forward_chain";
+
+/// The TcpForwardChainServer listens for incoming TCP connections from peers
+/// and spawns a client protocol handler for each accepted connection.
+/// It also implements the ForwardEventListener trait to receive forward events
+/// and forward them to all connected peers.
+#[derive(Clone)]
+pub struct TcpForwardChainServer {
+    our_tip: Arc<Mutex<HeaderTip>>,
+    clients: ActoRef<ClientMsg>,
+    _runtime: Arc<AcTokio>,
+}
+
+impl TcpForwardChainServer {
+    /// Creates a new TcpForwardChainServer instance:
+    ///
+    ///  - Start an Acto runtime
+    ///  - Bind a TCP listener to the given address
+    ///  - Spawn the client supervisor actor
+    #[expect(clippy::expect_used)]
+    #[expect(clippy::unwrap_used)]
```

Comment:
in production we should always use `.expect` to better mark the context of the error

Reply:
Makes sense, fixed.